webui/react/src/components/ResourceChart.tsx

```diff
@@ -0,0 +1,94 @@
+import Plotly, { Data, Layout } from 'plotly.js-basic-dist';
+import React from 'react';
+import createPlotlyComponent from 'react-plotly.js/factory';
+
+import { getStateColor, lightTheme } from 'themes';
+import { CommandState, CommonProps, Resource, ResourceState, resourceStates, resourceTypes } from 'types';
+
+const Plot = createPlotlyComponent(Plotly);
+interface Props extends CommonProps {
+  title: string;
+  resources: Resource[];
+}
+
+export interface PlotInfo {
+  data: Data[];
+  layout: Partial<Layout>;
+}
+
+const genPlotInfo = (title: string, resources: Resource[]): PlotInfo | null => {
+
+  const tallies: Record<ResourceState, number> = (() => {
+    const t: any = {};
+    resourceStates.forEach(state => (t[state]=0));
+    return t;
+  })();
+
+  resources.forEach(resource => {
+    if (!resource.container) {
+      tallies.FREE = tallies.FREE + 1;
+    } else {
+      tallies[resource.container.state] = tallies[resource.container.state]+1;
+    }
+  });
+
+  const labels: string[] = [];
+  const values: number[] = [];
+  const colors: string[] = [];
+  Object.entries(tallies).forEach(([ rsState, rsValue ]) => {
+    if (rsValue === 0) return;
+    labels.push(rsState);
+    values.push(rsValue);
+    if (rsState === ResourceState.Free) {
+      colors.push(lightTheme.colors.states.inactive);
```

Comment:
suggestion: is there a way to make this work with css targeting (css vars)? Otherwise when we want to change themes this will be hard coded to the light theme.

Reply:
i'll look futher into that but when we have multiple themes and users can choose the theme would we have a global state, maybe a context, where it would carry the user's choice? would that be able to return a `currentTheme` reference?

Reply:
The advantage of the CSS vars approach, if possible, is that it won't require JS to apply themes.

Reply:
that might prove difficult they call that out as something they probably won't support in Nov 2016. https://github.com/plotly/plotly.js/issues/1195#issuecomment-262793638

let me see if anything has changed.